File: pr.cpp
#include<bits/stdc++.h>
using namespace std ;

unsigned long long factorial(unsigned long long x){
    if (x == 1)
        return 1 ;
    return x*factorial(x-1);
}
int fun(unsigned long long n ,unsigned long long p){
      long long n_c = n - p;
      unsigned long  long f_p = factorial(p);
      unsigned long long temp = 0 ; 
      for(unsigned long long k = 2 ; k <= p ; k++){
          temp += f_p/(factorial(k)*(factorial(n - k))) ;
          if (temp > n_c) 
            return 0 ;
      }
      return 1 ;
}
int main(){
    int T ;
    cin>>T ;
    unsigned long long  n , p ; 
    while(T--){
        cin>>n>>p;
        cout<<fun(n - 1 ,p )<<endl;
    }
}

File: pr.py
from math import factorial as fac

def main():
    for i in range(int(input())):
        n , p = map(int , input().split())
        n -= 1
        n_c=  n - p
        f_p = fac(p)
        temp = 0 
        is_valid = True 
        for k in range(2,  p+1):
            temp += f_p/fac(k)*fac(n-k)
            if temp > n_c :
                is_valid = False
                print(0)
                break
        if is_valid : 
            print(1)

if __name__ == "__main__":
    main()
